client/db/src/lib.rs

```diff
@@ -358,10 +358,11 @@ impl DatabaseSource {
 	/// Return dabase path for databases that are on the disk.
 	pub fn path(&self) -> Option<&Path> {
 		match self {
-			DatabaseSource::Auto { .. } => {
-				// TODO:
-				unimplemented!();
-			},
+			// as per https://github.com/paritytech/substrate/pull/9500#discussion_r684312550
+			//
+			// IIUC this is needed for polkadot to create its own dbs, so until it can use parity db
+			// I would think rocksdb, but later parity-db.
+			DatabaseSource::Auto { paritydb_path, .. } => Some(&paritydb_path),
```

Comment:
I think this does not matter much for now. There's a separate item in #9201 to resolve polkadot db path properly.